asserts/snap_asserts.go

```diff
@@ -567,6 +567,11 @@ func (snaprev *SnapRevision) DeveloperID() string {
 	return snaprev.HeaderString("developer-id")
 }
 
+// Integrity returns the integrity data embedded in the snap revision if any.
+func (snaprev *SnapRevision) Integrity() interface{} {
```

Comment:
we might want to define a struct Integrity with fields and precache on the snaprev and return a pointer to that or nil, instead of the raw map, we don't have any other accesors except for the whole headers themselves that return interface{} atm

Reply:
I took a stab at it. If it looks good, I will need to update later PRs that I am using the old way to access those fields.